### ISSUE TITLE
Package causes failing tests

### ISSUE THREAD

**Issue by rtucek** [oliverfindl/vue-svg-inline-loader#37]
I've recently released a [query builder](https://github.com/rtucek/vue-query-builder) library and I'm using this package for inlining one svg file.
It works fine, however while preparing documentation and Travis CI, I've noticed that since I've added the package, all tests are failing with the error message:

>   ● Test suite failed to run
>
>    Cannot find module '/home/rtucek/projects/query-builder/node_modules/@vue/babel-preset-app/node_modules/@babel/runtime/helpers/interopRequireDefault' from 'QueryBuilder.vue'

I've been able to bisect the error to this commit (rtucek/vue-query-builder@bc037c1ee838266b566611f90345d035854bd0dc) - which is where I added this package.

**Comment by oliverfindl** [comment]
Hello,

I don't know what's causing your issue. This loader isn't using [@vue/babel-preset-app](https://github.com/vuejs/vue-cli/tree/dev/packages/@vue/babel-preset-app#readme). To be 100% sure, I ran grep in src, dist and node_modules with 0 matches for "babel-preset-app" string.

Error from your tests is from file [QueryBuilder.vue](https://github.com/rtucek/vue-query-builder/blob/c43c00b5b8eea8ee0092ef7bdc92fdd59157c79d/src/QueryBuilder.vue) and this file wasn't included in your [commit](https://github.com/rtucek/vue-query-builder/commit/bc037c1ee838266b566611f90345d035854bd0dc) mentioned above.

You can try to comment out this loader definition in your [vue.config.js](https://github.com/rtucek/vue-query-builder/blob/c43c00b5b8eea8ee0092ef7bdc92fdd59157c79d/vue.config.js#L13) and test if error is still occuring. If so, try to [inspect](https://cli.vuejs.org/guide/cli-service.html#vue-cli-service-inspect) [webpack.config.js](https://cli.vuejs.org/guide/webpack.html#inspecting-the-project-s-webpack-config) file, which is generated dynamically by [vue-cli](https://github.com/vuejs/vue-cli).
It can be also issue with corrupted node_modules. You can try to remove node_modules and package-lock.json and run fresh install.

Please give me feedback if this helps.

Thanks

**Comment by rtucek** [comment]
Hi,

thank you for looking into this!

> Error from your tests is from file QueryBuilder.vue and this file wasn't included in your commit mentioned above.

Yes, the tests haven been already there and passed successfully - until this package was added.

> It can be also issue with corrupted node_modules. You can try to remove node_modules and package-lock.json and run fresh install.

Already tried before opening the issue.

> You can try to comment out this loader definition in your vue.config.js and test if error is still occuring. If so, try to inspect webpack.config.js file, which is generated dynamically by vue-cli.

Unfortunately, there's nothing suspicious.

```diff
diff --git a/diff.json b/diff.json
index d80d870..a9bb753 100644
--- a/diff.json
+++ b/diff.json
@@ -51,33 +51,37 @@
     rules: [
       /* config.module.rule('vue') */
       {
         test: /\.vue$/,
         use: [
           /* config.module.rule('vue').use('cache-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/cache-loader/dist/cjs.js',
             options: {
               cacheDirectory: '/home/rtucek/projects/query-builder/node_modules/.cache/vue-loader',
-              cacheIdentifier: '575c2885'
+              cacheIdentifier: '562093ef'
             }
           },
           /* config.module.rule('vue').use('vue-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/vue-loader/lib/index.js',
             options: {
               compilerOptions: {
                 whitespace: 'condense'
               },
               cacheDirectory: '/home/rtucek/projects/query-builder/node_modules/.cache/vue-loader',
-              cacheIdentifier: '575c2885'
+              cacheIdentifier: '562093ef'
             }
+          },
+          /* config.module.rule('vue').use('vue-svg-inline-loader') */
+          {
+            loader: 'vue-svg-inline-loader'
           }
         ]
       },
       /* config.module.rule('images') */
       {
         test: /\.(png|jpe?g|gif|webp)(\?.*)?$/,
         use: [
           /* config.module.rule('images').use('url-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/url-loader/dist/cjs.js',
@@ -1068,21 +1072,21 @@
         test: /\.m?jsx?$/,
         exclude: [
           function () { /* omitted long function */ }
         ],
         use: [
           /* config.module.rule('js').use('cache-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/cache-loader/dist/cjs.js',
             options: {
               cacheDirectory: '/home/rtucek/projects/query-builder/node_modules/.cache/babel-loader',
-              cacheIdentifier: '400d2c47'
+              cacheIdentifier: '7e5fb1a6'
             }
           },
           /* config.module.rule('js').use('babel-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/babel-loader/lib/index.js'
           }
         ]
       },
       /* config.module.rule('eslint') */
       {
@@ -1104,21 +1108,21 @@
       },
       /* config.module.rule('ts') */
       {
         test: /\.ts$/,
         use: [
           /* config.module.rule('ts').use('cache-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/cache-loader/dist/cjs.js',
             options: {
               cacheDirectory: '/home/rtucek/projects/query-builder/node_modules/.cache/ts-loader',
-              cacheIdentifier: '5e2ab375'
+              cacheIdentifier: '30e4d60a'
             }
           },
           /* config.module.rule('ts').use('babel-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/babel-loader/lib/index.js'
           },
           /* config.module.rule('ts').use('ts-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/ts-loader/index.js',
             options: {
@@ -1133,21 +1137,21 @@
       },
       /* config.module.rule('tsx') */
       {
         test: /\.tsx$/,
         use: [
           /* config.module.rule('tsx').use('cache-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/cache-loader/dist/cjs.js',
             options: {
               cacheDirectory: '/home/rtucek/projects/query-builder/node_modules/.cache/ts-loader',
-              cacheIdentifier: '5e2ab375'
+              cacheIdentifier: '30e4d60a'
             }
           },
           /* config.module.rule('tsx').use('babel-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/babel-loader/lib/index.js'
           },
           /* config.module.rule('tsx').use('ts-loader') */
           {
             loader: '/home/rtucek/projects/query-builder/node_modules/ts-loader/index.js',
             options: {
```

**Comment by rtucek** [comment]
I think there's nothing wrong with the package itself.

With and without this package yarn picks @vue/babel-preset-app@4.1.2

However, without this package, I get this:
```sh
$ ls -A node_modules/@vue/babel-preset-app/node_modules/
@babel
```

...and with this package I have:
```sh
$ ls -A node_modules/@vue/babel-preset-app/node_modules/
core-js
```

I've no clue why this is happening :/

**Comment by oliverfindl** [comment]
Hello,

so to make it 100% sure, its get broken only by installing this loader (without actual usage)?

This seems to be some kind of issue / feature / bug with yarn or its lockfile (I'm not familiar with yarn, I'm using npm). You probably already tried this after your recent findings, but can you try to wipe out your node_modules and reinstall them with npm instead of yarn?

I noticed your project has issue with package [@vue/babel-preset-app](https://github.com/vuejs/vue-cli/tree/dev/packages/@vue/babel-preset-app#readme) which has [@babel/runtime](https://github.com/babel/babel/tree/master/packages/babel-runtime) as dependency. This loader uses this package too. So I wonder if yarn isn't trying to do some cross-referencing / symlinking magic with same packages and somehow silently failing due to user / group permissions, etc..

Please reply if using npm instead of yarn helps.

Thanks

**Comment by rtucek** [comment]
Hello,

> so to make it 100% sure, its get broken only by installing this loader (without actual usage)?

Exactly.

> You probably already tried this after your recent findings, but can you try to wipe out your node_modules and reinstall them with npm instead of yarn?

Yes, I tried that before opening the issue.

It's really odd.

**Comment by rtucek** [comment]
Alright, I've already found a workaround for my project, that doesn't require an external dependency for inlining the svg - just by tweaking the [vue.config.js](https://github.com/rtucek/vue-query-builder/pull/7/files#diff-0e893b3578ae3d5c8b98bac1c80756a7).

However, this is still interesting to debug.
Eventually, I found a fix by googling more towards babel related keywords and less in regards to Vue and found facebook/create-react-app#7183.
Adding @babel/runtime explicitly with `yarn add -D @babel/runtime` fixed it for me, however crosschecking with `npm install --save-dev @babel/runtime` still failed.

I'm closing this since my problem is fixed and the underlying error message is out of scope of this package.
Thank you for looking into it!
